tidy3d/components/base.py

```diff
@@ -922,6 +927,133 @@ def make_json_compatible(json_string: str) -> str:
         json_string = make_json_compatible(json_string)
         return json_string
 
+    def strip_traced_fields(
+        self, starting_path: tuple[str] = (), include_untraced_data_arrays: bool = False
+    ) -> AutogradFieldMap:
+        """Extract a dictionary mapping paths in the model to the data traced by ``autograd``.
+
+        Parameters
+        ----------
+        starting_path : tuple[str, ...] = ()
+            If provided, starts recursing in self.dict() from this path of field names
+        include_untraced_data_arrays : bool = False
+            Whether to include ``DataArray`` objects without tracers.
+            We need to include these when returning data, but are unnecessary for structures.
+
+        Returns
+        -------
+        dict
+            mapping of traced fields used by ``autograd``
+
+        """
+
+        field_mapping = {}
+
+        def handle_value(x: Any, path: tuple[str, ...]) -> None:
```

Comment:
I don't understand actually, sorry but could you write the function for me if you think it will improve speed? (we can also leave for later if you want)

Reply:
it worked! but was about 5% slower than the original :/

Reply:
oh damn :(